.github/workflows/build.yml

```diff
@@ -0,0 +1,56 @@
+name: "Demo Build Checks"
+
+env:
+  bashPass: \033[32;1mPASSED -
+  bashInfo: \033[33;1mINFO -
+  bashFail: \033[31;1mFAILED -
+  bashEnd:  \033[0m
+
+on:
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
```suggestion
    strategy:
      fail-fast: false
      matrix:
```

Reply:
Allow all jobs to run even if a single one fails

Reply:
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#handling-failures for more info

Reply:
https://github.com/FreeRTOS/iot-reference-esp32c3/actions/runs/7216197357 example of diff